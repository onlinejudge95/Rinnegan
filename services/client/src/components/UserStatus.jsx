import React from "react";
import axios from "axios";
import PropTypes from "prop-types";

class UserStatus extends React.Component {
  state = { email: "", username: "" };

  componentDidMount() {
    this.getUserStatus();
  }

  getUserStatus = (event) => {
    const headers = {
      Accept: "application/json",
      Authorization: `Bearer ${this.props.accessToken}`,
      "Content-Type": "application/json",
    };
    return axios
      .get(`${process.env.REACT_APP_USERS_SERVICE_URL}/auth/status`, {
        headers,
      })
      .then((response) => {
        this.setState({
          email: response.data.email,
          username: response.data.username,
        });
      })
      .catch((err) => {
        console.log(err);
      });
  };

  render() {
    return (
      <div>
        <ul>
          <li>
            <strong>Email:</strong>&nbsp;
            <span data-testid="user-email">{this.state.email}</span>
          </li>
          <li>
            <strong>Username:</strong>&nbsp;
            <span data-testid="user-username">{this.state.username}</span>
          </li>
        </ul>
      </div>
    );
  }
}

UserStatus.propTypes = {
  accssToken: PropTypes.string,
};

export default UserStatus;
