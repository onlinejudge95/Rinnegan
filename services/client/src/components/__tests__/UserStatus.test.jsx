import React from "react";
import { render, cleanup, wait } from "@testing-library/react";
import axios from "axios";

import UserStatus from "../UserStatus";
import { Route } from "react-router-dom";

afterEach(cleanup);

jest.mock("axios");

axios.get.mockImplementation(() =>
  Promise.resolve({
    data: { email: "test_user@mail.com", username: "test_user" },
  })
);

const mockProps = {
  isAuthenticated: () => {
    return true;
  },
};

it("renders properly when authenticated", async () => {
  const dom = render(<UserStatus {...mockProps} />);

  await wait(() => {
    expect(axios.get).toHaveBeenCalledTimes(1);
  });

  expect((await dom.findByTestId("user-email")).innerHTML).toBe(
    "test_user@mail.com"
  );
  expect((await dom.findByTestId("user-username")).innerHTML).toBe("test_user");
});

it("renders", async () => {
  const dom = renderWithRouter(<UserStatus {...mockProps} />);

  await wait(() => {
    expect(axios.get).toHaveBeenCalled();
  });

  expect(dom.asFragment()).toMatchSnapshot();
});

// TODO:- Write test for validating a refresh token if it has been altered and a new tab is opened with the status Route, it should redirect to login route
