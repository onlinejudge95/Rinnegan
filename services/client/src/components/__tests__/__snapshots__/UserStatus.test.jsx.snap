// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <ul>
      <li>
        <strong>
          Email:
        </strong>
         
        <span
          data-testid="user-email"
        >
          test_user@mail.com
        </span>
      </li>
      <li>
        <strong>
          Username:
        </strong>
         
        <span
          data-testid="user-username"
        >
          test_user
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
